File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    memory = {}
    answer = []

    for n in name:
        memory[n] = 0
    i = 0
    for key in memory:
        memory[key] = yearning[i]
        i += 1
            
    for p in photo:
        m_num = 0
        for n in p:
            if n in memory:
                m_num += memory[n]
        answer.append(m_num)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photos) {
    vector<int> answer;
    unordered_map<string, int> yearningMap;
    for (int i = 0; i < name.size(); i++) {
        yearningMap[name[i]] = yearning[i];
    }

    for (const auto &photo : photos) {
        int cnt = 0;
        for (const auto &name: photo) {
            if (yearningMap.find(name) != yearningMap.end()) cnt += yearningMap[name];
        }
        answer.push_back(cnt);
    }
    return answer;
}


